src/mint/library.cairo

```diff
@@ -296,4 +322,56 @@ namespace CarbonableMinter:
         mint_n(nft_contract_address, to, token_id, new_quantity)
         return ()
     end
+
+    ###
+    # Verify that leaf belongs to merkle tree.
+    # ref: https://github.com/ncitron/cairo-merkle-distributor/blob/master/contracts/merkle.cairo
+    # @param leaf
+    # @param proof_len
+    # @param proof
+    ###
+    func merkle_verify{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
```

Comment:
I suggest we more this function and `calc_merkle_root` in a separate file, can be done in a follow up PR

Reply:
Done in [98223d1](https://github.com/Carbonable/carbonable-starknet/pull/22/commits/98223d10be7c60a44e759013f91cd29b433bbaeb)